x-pack/plugin/ml/qa/native-multi-node-tests/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/PyTorchModelIT.java

```diff
@@ -111,21 +117,30 @@ public void unsetLogging() throws IOException {
         RAW_MODEL_SIZE = Base64.getDecoder().decode(BASE_64_ENCODED_MODEL).length;
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/pull/75759")
-    public void testEvaluate() throws IOException {
+    public void testEvaluate() throws Exception {
         createModelStoreIndex();
         putTaskConfig();
         putModelDefinition();
         refreshModelStoreIndex();
         createTrainedModel();
         startDeployment();
+        CountDownLatch latch = new CountDownLatch(10);
         try {
             // Adding multiple inference calls to verify different calls get routed to separate nodes
             for (int i = 0; i < 10; i++) {
-                Response inference = infer("my words");
-                assertThat(EntityUtils.toString(inference.getEntity()), equalTo("{\"inference\":[[1.0,1.0]]}"));
+                executorService.execute(() -> {
+                    try {
+                        Response inference = infer("my words");
+                        assertThat(EntityUtils.toString(inference.getEntity()), equalTo("{\"inference\":[[1.0,1.0]]}"));
+                    } catch (IOException ex) {
+                        fail(ex.getMessage());
```

Comment:
this won't fail the test as it is not thrown from the main test thread. To make this work collect the responses and exceptions in a list and assert on them after the countdown.

Reply:
I was wrong about this the assertion from the executor thread does propagate and fails the test.

This test fails 5 times

```Java
    public void testThreadAssert() throws InterruptedException {
        ExecutorService executorService = Executors.newFixedThreadPool(5);
        CountDownLatch latch = new CountDownLatch(5);

        for (int i = 0; i < 5; i++) {
            executorService.execute(() -> {
                    latch.countDown();
                    fail("fail from thread");
            });
        }

        latch.await();
    }
```